API/Attributes/RequireGroupMembershipAttribute.cs

```diff
@@ -0,0 +1,35 @@
+using API.Extensions;
+using Application.Exceptions;
+using Application.Interfaces;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace API.Attributes;
+
+/// <summary>
+/// Attribute that ensures the user is a member of the group before executing the action.
+/// </summary>
+[AttributeUsage(AttributeTargets.Method)]
+public class RequireGroupMembershipAttribute : Attribute, IAsyncActionFilter
+{
+    /// <summary>
+    /// Checks if the user is authorized to access the group.
+    /// </summary>
+    /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+    /// <param name="context">The context for the action being executed.</param>
+    /// <param name="next">The delegate to execute the next action filter or the action itself.</param>
+    public async Task OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next)
+    {
+        if (!context.RouteData.Values.TryGetValue("id", out var groupIdObj) ||
+            !int.TryParse(groupIdObj?.ToString(), out var groupId))
+        {
+            throw new InternalServerErrorException();
+        }
+
```

Comment:
Lepiej użyć `nameof` czy string.
Zamiast walidować i zwracać 500, można po prostu założyć że wszystko jest git. Jak wywali jakikolwiek błąd, to Middleware to przechwyci i zwróci 500.

```suggestion
        var groupId = int.Parse(context.RouteData.Values[nameof(Group.Id)]!.ToString()!);
```